### PR TITLE
Improve autocomplete behaviour

### PR DESCRIPTION
**Pull request by aviraldg**
Fixes vector-im/vector-web#1761

### CODE REVIEW

File: src/components/views/rooms/Autocomplete.js [hunk 1]
```diff
@@ -19,79 +25,138 @@ export default class Autocomplete extends React.Component {
...
+        // selectionOffset = 0, so we don't end up completing when autocomplete is hidden
+        this.setState({hide: true, selectionOffset: 0});
+    }
+
+    forceComplete() {
+        this.setState({
+            forceComplete: true,
+        }, () => {
+            this.complete(this.props.query, this.props.selection);
+            setTimeout(() => this.onDownArrow(), 50); // FIXME HACK
```
Comment (ara4n) [inline src/components/views/rooms/Autocomplete.js]: why is this needed?

**Comment by ara4n** [comment]
lgtm other than:
 * question on setTimeout hack
 * `Mat<tab><tab>` doesn't autocomplete to Matthew - it just selects Matthew in the autocompleter but doesn't actually apply it to the composer input
 * Rooms without aliases should not be included in the autocomplete results at all.

**Comment by aviraldg** [comment]
setTimeout hack -- was trying to avoid more promises, but apperntly not.
Mat<tab> should work now
Rooms without aliases should no longer appear
